core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -108,30 +80,20 @@ public TableScan asOfTime(long timestampMillis) {
     }
 
     @Override
-    protected CloseableIterable<FileScanTask> planFiles(
-        TableOperations ops, Snapshot snapshot, Expression rowFilter,
-        boolean ignoreResiduals, boolean caseSensitive, boolean colStats) {
-      CloseableIterable<ManifestFile> manifests = allDataManifestFiles(
-          ops.current().snapshots(), context().planExecutor());
-      String schemaString = SchemaParser.toJson(schema());
-      String specString = PartitionSpecParser.toJson(PartitionSpec.unpartitioned());
-      Expression filter = ignoreResiduals ? Expressions.alwaysTrue() : rowFilter;
-      ResidualEvaluator residuals = ResidualEvaluator.unpartitioned(filter);
-
-      return CloseableIterable.transform(manifests, manifest ->
-          new ManifestReadTask(ops.io(), ops.current().specsById(), manifest, schema(),
-              schemaString, specString, residuals));
+    public CloseableIterable<FileScanTask> planFiles() {
+      return super.planAllFiles();
     }
-  }
 
-  private static CloseableIterable<ManifestFile> allDataManifestFiles(
-      List<Snapshot> snapshots, ExecutorService workerPool) {
-    try (CloseableIterable<ManifestFile> iterable = new ParallelIterable<>(
-        Iterables.transform(snapshots, snapshot -> (Iterable<ManifestFile>) () -> snapshot.dataManifests().iterator()),
-        workerPool)) {
-      return CloseableIterable.withNoopClose(Sets.newHashSet(iterable));
-    } catch (IOException e) {
-      throw new RuntimeIOException(e, "Failed to close parallel iterable");
+    @Override
+    protected CloseableIterable<ManifestFile> manifests() {
```

Comment:
We change the return type of manifests to CloseableIterable for this table? I just notice all the other implementations just wrap the list. I think this makes sense

Reply:
Yea as the AllDataFilesTable used to use a parallel iterable.  It's more flexible I think in case the other Files tables want to use that later.

Reply:
Seems reasonable.